cpp/src/arrow/dataset/dataset.h

```diff
@@ -52,12 +52,21 @@ class ARROW_DS_EXPORT DataFragment {
 
   virtual ~DataFragment() = default;
 
+  /// \brief An expression which evaluates to true for all data viewed by this
+  /// DataFragment. May be null, which indicates no information is available.
```

Comment:
It ain't null, it would be `scalar(true)`. Maybe we should opt for optional instead, as middle ground between no information and not nullptr.

Reply:
I'll correct the comment